### PR TITLE
Allow estimator to be set for MGXS

### PR DESCRIPTION
**Pull request by paulromano**
This short PR enables users to set the estimator type for MGXS classes. @wbinventor If you have any suggestions for how we might change `Library`, please feel free to share.

### CODE REVIEW

File: openmc/mgxs/mgxs.py [hunk 1]
```diff
@@ -47,6 +47,10 @@
                 'material',
                 'mesh']
 
+ESTIMATOR_TYPES = ['tracklength',
```
Comment (wbinventor) [inline openmc/mgxs/mgxs.py]: Perhaps this should be defined in "tally.py"?

File: openmc/mgxs/mgxs.py [hunk 2]
```diff
@@ -2190,7 +2196,7 @@ class TransportXS(MGXS):
     tally_keys : list of str
         The keys into the tallies dictionary for each tally used to compute
         the multi-group cross section
-    estimator : {'tracklength', 'analog'}
+    estimator : {'tracklength', 'collision', 'analog'}
```
Comment (wbinventor) [inline openmc/mgxs/mgxs.py]: I'm confused about this one - this cannot be tracklength right? If so, shouldn't the docstring reflect that the tallies can only be analog or collision?

Comment (wbinventor) [inline openmc/mgxs/mgxs.py]: I'm confused about this one - this cannot be tracklength right? If so, shouldn't the docstring reflect that the tallies can only be analog or collision?

File: openmc/mgxs/mgxs.py [hunk 3]
```diff
@@ -2320,7 +2323,7 @@ class NuTransportXS(TransportXS):
...
```
Comment (wbinventor) [inline openmc/mgxs/mgxs.py]: And the same comment here, as well as for `ScatterMatrixXS` and its subclasses.

File: openmc/mgxs/mgxs.py [hunk 4]
```diff
@@ -368,6 +373,11 @@ def nuclides(self, nuclides):
         cv.check_iterable_type('nuclides', nuclides, basestring)
         self._nuclides = nuclides
 
+    @estimator.setter
+    def estimator(self, estimator):
```
Comment (wbinventor) [inline openmc/mgxs/mgxs.py]: If my comment about restricting the estimator to only analog and collision for some MGXS is well founded, then this property should check for only that subset of types for those subclasses.  Perhaps this property needs to be redefined a few places, or there could be a new `_accepted_types` class attribute for type checking.

**Comment by wbinventor** [comment]
@paulromano this looks very straightforward aside from my comment about restricted types. Setting the estimators at the `Library` level is certainly more tricky. One idea would be a `Library.set_estimators(mgxs_type, estimator)` to assign estimators for each MGXS type. This fine-grained approach might be compressed a bit if the input parameter were instead a dictionary of MGXS types mapped to estimators. What do you think?

**Comment by paulromano** [comment]
@wbinventor I opted to just have a `Library.estimator` property that can be set prior to calling `build_library()`. By default, it is set to `None`, meaning it won't override whatever the default is for each MGXS type. However, you can set it to "analog", which will apply to each MGXS that it creates. That is really the only use case I can think of right now so I think it should be ok. If a user really wants more fine-grained control, they can always iterate over `all_mgxs` and make modifications themselves.

Let me know if you're satisfied with the PR as is now.

**Comment by wbinventor** [comment]
Looks fine to me.

**Pull request merged**